src/arkode/arkode_lsrkstep_io.c

```diff
@@ -0,0 +1,661 @@
+    step_mem->req_stages        = 10;
+    step_mem->nfusedopvecs      = 3;
+    step_mem->q = ark_mem->hadapt_mem->q = 2;
+    step_mem->p = ark_mem->hadapt_mem->p = 1;
+    break;
+  case ARKODE_LSRK_SSP_S_3:
+    ark_mem->step               = lsrkStep_TakeStepSSPs3;
+    step_mem->LSRKmethod        = ARKODE_LSRK_SSP_S_3;
+    step_mem->is_SSP            = SUNTRUE;
+    ark_mem->step_printallstats = lsrkSSPStep_PrintAllStats;
+    step_mem->req_stages        = 9;
+    step_mem->nfusedopvecs      = 3;
+    step_mem->q = ark_mem->hadapt_mem->q = 3;
+    step_mem->p = ark_mem->hadapt_mem->p = 2;
+    break;
+  case ARKODE_LSRK_SSP_10_4:
+    ark_mem->step               = lsrkStep_TakeStepSSP104;
+    step_mem->LSRKmethod        = ARKODE_LSRK_SSP_10_4;
+    step_mem->is_SSP            = SUNTRUE;
+    ark_mem->step_printallstats = lsrkSSPStep_PrintAllStats;
+    step_mem->req_stages        = 10;
+    step_mem->nfusedopvecs      = 3;
+    step_mem->q = ark_mem->hadapt_mem->q = 4;
+    step_mem->p = ark_mem->hadapt_mem->p = 3;
+    break;
+
+  default:
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "Invalid method option.");
+    return (ARK_ILL_INPUT);
+  }
+
+  return (ARK_SUCCESS);
+}
+
+int LSRKStepSetMethodByName(void* arkode_mem, const char* emethod)
+{
+  if (strcmp(emethod, "ARKODE_LSRK_RKC_2") == 0)
+  {
+    return (LSRKStepSetMethod(arkode_mem, ARKODE_LSRK_RKC_2));
+  }
+  if (strcmp(emethod, "ARKODE_LSRK_RKL_2") == 0)
+  {
+    return (LSRKStepSetMethod(arkode_mem, ARKODE_LSRK_RKL_2));
+  }
+  if (strcmp(emethod, "ARKODE_LSRK_SSP_S_2") == 0)
+  {
+    return (LSRKStepSetMethod(arkode_mem, ARKODE_LSRK_SSP_S_2));
+  }
+  if (strcmp(emethod, "ARKODE_LSRK_SSP_S_3") == 0)
+  {
+    return (LSRKStepSetMethod(arkode_mem, ARKODE_LSRK_SSP_S_3));
+  }
+  if (strcmp(emethod, "ARKODE_LSRK_SSP_10_4") == 0)
+  {
+    return (LSRKStepSetMethod(arkode_mem, ARKODE_LSRK_SSP_10_4));
+  }
+
+  arkProcessError(NULL, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                  "Unknown method type");
+
+  return ARK_ILL_INPUT;
+}
+
+/*---------------------------------------------------------------
+  LSRKStepSetDomEigFn specifies the dom_eig function.
+  ---------------------------------------------------------------*/
+int LSRKStepSetDomEigFn(void* arkode_mem, ARKDomEigFn dom_eig)
+{
+  ARKodeMem ark_mem;
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeMem and ARKodeLSRKStepMem structures */
+  retval = lsrkStep_AccessARKODEStepMem(arkode_mem, __func__, &ark_mem,
+                                        &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  /* set the dom_eig routine pointer, and update relevant flags */
+  if (dom_eig != NULL)
+  {
+    step_mem->is_ext_dom_eig = SUNTRUE;
+    step_mem->extDomEig      = dom_eig;
+
+    return (ARK_SUCCESS);
+  }
+  else
+  {
+    step_mem->is_ext_dom_eig = SUNFALSE;
+    step_mem->extDomEig      = NULL;
```

Comment:
There should be a check (at least in `lsrkStep_Init`) that `dom_eig_fn` is not `NULL`

Reply:
If the check is done at initialization then it would only need to happen once instead of every eigenvalue update

Reply:
Agreed; it would be best if we just perform this check at initialization.